sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventDataForm.java

```diff
@@ -325,10 +332,33 @@ protected void addFields() {
 		locationForm.setFacilityFieldsVisible(getField(EventDto.TYPE_OF_PLACE).getValue() == TypeOfPlace.FACILITY, true);
 		typeOfPlace.addValueChangeListener(e -> locationForm.setFacilityFieldsVisible(e.getProperty().getValue() == TypeOfPlace.FACILITY, true));
 
+		regionField.addValueChangeListener(e -> {
+			RegionReferenceDto region = (RegionReferenceDto) regionField.getValue();
+			if (region != null) {
+				responsibleUserSurveillanceSupervisors =
+					FacadeProvider.getUserFacade().getUsersByRegionAndRoles(region, UserRole.SURVEILLANCE_SUPERVISOR);
+				responsibleUserSurveillanceOfficers = FacadeProvider.getUserFacade().getUsersByRegionAndRoles(region, UserRole.SURVEILLANCE_OFFICER);
+			} else {
+				responsibleUserSurveillanceSupervisors.clear();
+				responsibleUserSurveillanceOfficers.clear();
+			}
+		});
+
 		districtField.addValueChangeListener(e -> {
-			List<UserReferenceDto> assignableSurveillanceOfficers = FacadeProvider.getUserFacade()
-				.getUserRefsByDistrict((DistrictReferenceDto) districtField.getValue(), false, UserRole.SURVEILLANCE_OFFICER);
-			FieldHelper.updateItems(surveillanceOfficerField, assignableSurveillanceOfficers);
+			DistrictReferenceDto district = (DistrictReferenceDto) districtField.getValue();
+			if (district != null) {
+				List<UserReferenceDto> currentDistrictSurveillanceOfficers =
+					FacadeProvider.getUserFacade().getUserRefsByDistrict(district, false, UserRole.SURVEILLANCE_OFFICER);
+
+				Set<UserReferenceDto> responsibleUsers = new LinkedHashSet<>();
+				responsibleUsers.addAll(currentDistrictSurveillanceOfficers);
```

Comment:
This logic doesn't really make sense; the field would either be populated by all surveillance officers from the region if the district filter is empty, or with all surveillance officers from the district if it's not empty, but not with both (because that would make the district filter completely redundant).

Reply:
That's what I've been asked to do.

Quoting Bernard :
> Order the names with surveillance officers in the district of the event at top of list, other SO and then supervisors follow.

The sole purpose of this is to put the district's SO on top of the list.

Reply:
I see - I will ask Bernard what the purpose of this specification is because I don't think it makes sense, and get back to you once he's responded.

Reply:
@alexandre-gille Verified with Bernard that the specification was not correct. The only users added to the dropdown to select the responsible users should be surveillance supervisors from the event's region and surveillance officers from the district's region.